library/core/src/intrinsics/mod.rs

```diff
@@ -4537,6 +4537,32 @@ pub(crate) const fn miri_promise_symbolic_alignment(ptr: *const (), align: usize
     )
 }
 
+#[requires(crate::mem::size_of::<T>() >= crate::mem::size_of::<U>())] //U cannot be larger than T
+#[ensures(|ret: &U| (ret as *const U as usize) % crate::mem::align_of::<U>() == 0)] //check that the output has expected alignment
+pub unsafe fn transmute_unchecked_wrapper<T,U>(input: T) -> U {    
```

Comment:
Can you please add a comment on why we need this and make this function private?